### PR TITLE
Add isCompressed flag to Blob Daat structure.

### PR DESCRIPTION
**Pull request by panchow**
Splitting the PR into multiple small PRs.

This PR adds isCompressed field to Data Blob structure.
Other minor cleanup changes.

### CODE REVIEW

File: ambry-messageformat/src/test/java/com/github/ambry/messageformat/BlobDataTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    BlobData data = new BlobData(BlobType.MetadataBlob, 100, buf);
+    Assert.assertEquals(BlobType.MetadataBlob, data.getBlobType());
+    Assert.assertEquals(100, data.getSize());
+    Assert.assertFalse(data.isCompressed());
+
+    // Test the other constructor.
+    data = new BlobData(BlobType.MetadataBlob, 100, buf, true);
+    Assert.assertTrue(data.isCompressed());
+  }
+}
```
Comment (snalli) [inline ambry-messageformat/src/test/java/com/github/ambry/messageformat/BlobDataTest.java:35]: new line

**Review by snalli — CHANGES_REQUESTED** [review]
add newline

**Comment by snalli** [comment]
Blob Daat -> typo in title

**Comment by justinlin-linkedin** [comment]
why is this merged already?

**Pull request merged**